docs/src/main/asciidoc/spring-integration-pubsub.adoc

```diff
@@ -156,7 +156,36 @@ public void messageReceiver(String payload,
 }
 ----
 
-NOTE: `AcknowledgeablePubSubMessage` objects acquired by synchronous pull are aware of their own acknowledgement IDs.
+Messages can only be retrieved one at a time from a `PubSubMessageSource`. To retrieve a batch of messages, use `PubSubBatchMessageSource`.
+
+[source,java]
+----
+@Bean
+@InboundChannelAdapter(channel = "pubsubInputChannel", poller = @Poller(fixedDelay = "100"))
+public MessageSource<Object> pubsubAdapter(PubSubTemplate pubSubTemplate) {
+	PubSubBatchMessageSource messageSource = new PubSubBatchMessageSource(pubSubTemplate, "exampleSubscription");
+	messageSource.setAckMode(AckMode.MANUAL);
+	messageSource.setBlockOnPull(true);
+	messageSource.setMaxFetchSize(100);
+	return messageSource;
+}
+----
+
+Payload type is not configurable for `PubSubBatchMessageSource`, since batched messages are delivered as a list of `AcknowledgeablePubsubMessage` objects.
+
+[source,java]
+----
+@ServiceActivator(inputChannel = "pubsubInputChannel")
+public void messageReceiver(List<AcknowledgeablePubsubMessage> payload) {
```

Comment:
I think we still need to document the `ACKNOWLEDGMENT_CALLBACK` header here since it can be used to ack/nack the whole batch.

Reply:
As discussed, I think spring-gcp team is in a better place to provide documentation that meets standards.